PROPOSAL-DRAFT-V1.0.md

```diff
@@ -0,0 +1,279 @@
+
+For example, Consider the following object Schema of User:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "age": {
+      "type": "number"
+    },
+    "email": {
+      "type": "string"
+    },
+    "address": {
+      "type": "object",
+      "properties": {
+        "street": {
+          "type": "string"
+        },
+        "city": {
+          "type": "string"
+        },
+        "country": {
+          "type": "string"
+        },
+        "zipcode": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+```
+The above schema has a lot of repetition and if the schema is more complex that have nested object or array of object, it gets more complex to write.
+
+On the other hand, with TSSG, above schema can be written as:
+```javascript
+{
+    name: string,
+    age: number,
+    email: string,
+    address: {
+        street: string,
+        city: string,
+        country: string,
+        zipcode: string,
+    }
+}
+
+```
+
+
+
+## Language
+A TSSG document is defined as a syntactic grammar where terminal symbols are tokens (indivisible lexical units). These tokens are defined in a lexical grammar which matches patterns of source characters.
+This sequence of lexical tokens are then scanned from left to right to produce an abstract syntax tree (AST) according to the Document syntactical grammar.
+
+
+We refer to A TSSG documents as programs. A program may contain expression blocks(schemas, requestbodies, paths, parameters), data types and Ignored lexical grammars(comments, whitespaces, line terminators).
+
+### White Spaces
+White space is used to improve legibility of source text and act as separation between tokens, and any amount of white space may appear before or after any token. White space between tokens is not significant to the semantic meaning of a TSSG Document, however white space characters may appear within a String or Comment token.
+
+### Line Terminators
+
+Like white space, line terminators are used to improve the legibility of source text, any amount may appear before or after any other token and have no significance to the semantic meaning of a TSSG Document. Line terminators are not found within any other token.
+
+
+### Comments
+TSSG source documents may contain multi‐line comments, starting with the `/*` marker and ending with `*/` marker.
+
+A comment can contain any Unicode code point in SourceCharacter except LineTerminator so a comment always consists of all code points starting with the `/*` character up to `*/`
+
+Comments are Ignored like white space and may appear after any token, or before a LineTerminator, and have no significance to the semantic meaning of a TSSG Document.
+```javascript
+/* this is
+multi line 
+comment */
+```
+### Data Types
```

Comment:
Please call this lexical tokens  @tauqeernasir correct me if I am wrong.

Reply:
Yes, they can also be call `lexical tokens`, @yogeshkathayat please update this as well.